https://savannah.gnu.org/bugs/index.php?52076

commit faa0098ac6b196375b00c48528a91ee553313e8d
Author: Bernhard M. Wiedemann <bwiedemann@suse.de>
Date:   Tue Sep 5 13:16:08 2017 +0200

    Sort glob file lists
    
    to make builds more reproducible by default.
    See https://reproducible-builds.org/ for why this is good.
    
    commit b9f831b8 added GLOB_NOSORT for no apparent reason,
    possibly by mistake.
    
    man 3 glob  is very specific that the only reason to add this flag is
    to save processing time, but since ordering actually matters in a
    variety of cases, (e.g. when linking .o files into a binary,
    or appending snippets to a collection)
    we want the list sorted.

Index: make-4.2.1/read.c
===================================================================
--- make-4.2.1.orig/read.c
+++ make-4.2.1/read.c
@@ -3275,7 +3275,7 @@ parse_file_seq (char **stringp, unsigned
           nlist = &name;
         }
       else
-        switch (glob (name, GLOB_NOSORT|GLOB_ALTDIRFUNC, NULL, &gl))
+        switch (glob (name, GLOB_ALTDIRFUNC, NULL, &gl))
           {
           case GLOB_NOSPACE:
             OUT_OF_MEM();
